cmd/buck/cli/archive.go

```diff
@@ -35,21 +35,30 @@ var defaultArchiveConfigCmd = &cobra.Command{
 }
 
 var setDefaultArchiveConfigCmd = &cobra.Command{
-	Use:     "set-default-config [(optional)file]",
-	Short:   "Set the default archive storage configuration for the specified Bucket.",
-	Long:    `Set the default archive storage configuration for the specified Bucket from a file, stdin, or options.`,
+	Use:   "set-default-config [(optional)file]",
+	Short: "Set the default archive storage configuration for the specified Bucket.",
+	Long: `Set the default archive storage configuration for the specified Bucket from a file, stdin, or flags.
+
+If flags are specified, this command updates the current default storage-config with the *explicitely set* flags. 
+Flags that aren't explicitely set won't set the default value, and thus keep the original value in the storage-config.
+
+If a file or stdin is used, the storage-config will be completely overriden by the provided one.`,
 	Example: "hub buck archive set-default-config --rep-factor=3 --fast-retrieval --verified-deal --trusted-miners=f08240,f023467,f09848",
 	Args:    cobra.RangeArgs(0, 1),
 	Run: func(c *cobra.Command, args []string) {
-		config := local.ArchiveConfig{}
-		stdIn, err := c.Flags().GetBool("")
+		var config local.ArchiveConfig
+		conf, err := bucks.NewConfigFromCmd(c, ".")
+		cmd.ErrCheck(err)
+
+		stdIn, err := c.Flags().GetBool("stdin")
```

Comment:
stdin flag name was missing.

Reply:
ah no, this _was_ right i think because it was accepting a flag called `--` or -i`. so `GetBool("")` would grab that naked double hyphen. but if changed to `stdin`, makes sense now